BuildResidentialHPXML/measure.rb

```diff
@@ -0,0 +1,3573 @@
+    hot_water_distribution_system_type_choices << "Standard"
+    hot_water_distribution_system_type_choices << "Recirculation"
+
+    arg = OpenStudio::Measure::OSArgument::makeChoiceArgument("hot_water_distribution_system_type", hot_water_distribution_system_type_choices, true)
+    arg.setDisplayName("Hot Water Distribution: System Type")
+    arg.setDescription("The type of the hot water distribution system.")
+    arg.setDefaultValue("Standard")
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("standard_piping_length", true)
+    arg.setDisplayName("Hot Water Distribution: Standard Piping Length")
+    arg.setUnits("ft")
+    arg.setDescription("The length of the standard piping.")
+    arg.setDefaultValue(50)
+    args << arg
+
+    recirculation_control_type_choices = OpenStudio::StringVector.new
+    recirculation_control_type_choices << "no control"
+    recirculation_control_type_choices << "timer"
+    recirculation_control_type_choices << "temperature"
+    recirculation_control_type_choices << "presence sensor demand control"
+    recirculation_control_type_choices << "manual demand control"
+
+    arg = OpenStudio::Measure::OSArgument::makeChoiceArgument("recirculation_control_type", recirculation_control_type_choices, true)
+    arg.setDisplayName("Hot Water Distribution: Recirculation Control Type")
+    arg.setDescription("The type of hot water recirculation control, if any.")
+    arg.setDefaultValue("no control")
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("recirculation_piping_length", true)
+    arg.setDisplayName("Hot Water Distribution: Recirculation Piping Length")
+    arg.setUnits("ft")
+    arg.setDescription("The length of the recirculation piping.")
+    arg.setDefaultValue(50)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("recirculation_branch_piping_length", true)
+    arg.setDisplayName("Hot Water Distribution: Recirculation Branch Piping Length")
+    arg.setUnits("ft")
+    arg.setDescription("The length of the recirculation branch piping.")
+    arg.setDefaultValue(50)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("recirculation_pump_power", true)
+    arg.setDisplayName("Hot Water Distribution: Recirculation Pump Power")
+    arg.setUnits("W")
+    arg.setDescription("The power of the recirculation pump.")
+    arg.setDefaultValue(50)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument("hot_water_distribution_pipe_r_value", true)
+    arg.setDisplayName("Hot Water Distribution: Insulation Nominal R-Value")
+    arg.setUnits("h-ft^2-R/Btu")
+    arg.setDescription("Nominal R-value of the insulation on the DHW distribution system.")
+    arg.setDefaultValue(0.0)
+    args << arg
+
+    dwhr_facilities_connected_choices = OpenStudio::StringVector.new
+    dwhr_facilitie
```

Comment:
Either remove this input or add description that if it's provided, it overrides all the other Solar Thermal inputs (and then implement that logic).